Mathlib/FieldTheory/LinearDisjoint.lean

```diff
@@ -365,6 +365,79 @@ theorem of_finrank_sup [FiniteDimensional F A] [FiniteDimensional F B]
     (H : finrank F ↥(A ⊔ B) = finrank F A * finrank F B) : A.LinearDisjoint B :=
   linearDisjoint_iff'.2 <| .of_finrank_sup_of_free (by rwa [← sup_toSubalgebra_of_left])
 
+/-- If `A` and `L` are linearly disjoint, one of them is algebraic, then `[L(A) : L] = [A : F]`. -/
+theorem adjoin_rank_eq_rank_left_of_isAlgebraic (H : A.LinearDisjoint L)
+    (halg : Algebra.IsAlgebraic F A ∨ Algebra.IsAlgebraic F L) :
+    Module.rank L (adjoin L (A : Set E)) = Module.rank F A := by
+  refine Eq.trans ?_ (Subalgebra.LinearDisjoint.adjoin_rank_eq_rank_left H)
+  set L' := (IsScalarTower.toAlgHom F L E).range
+  let i : L ≃ₐ[F] L' := AlgEquiv.ofInjectiveField (IsScalarTower.toAlgHom F L E)
+  have heq : (adjoin L (A : Set E)).toSubalgebra.toSubsemiring =
+      (Algebra.adjoin L' (A : Set E)).toSubsemiring := by
+    rw [adjoin_toSubalgebra_of_isAlgebraic _ _ halg.symm, Algebra.adjoin_toSubsemiring,
+      Algebra.adjoin_toSubsemiring]
+    congr 2
+    ext x
+    simp only [Set.mem_range, Subtype.exists]
+    exact ⟨fun ⟨y, h⟩ ↦ ⟨x, ⟨y, h⟩, rfl⟩, fun ⟨a, ⟨y, h1⟩, h2⟩ ↦ ⟨y, h1.trans h2⟩⟩
+  refine rank_eq_of_equiv_equiv i (RingEquiv.subsemiringCongr heq).toAddEquiv
+    i.bijective fun a ⟨x, hx⟩ ↦ ?_
+  ext
+  simp_rw [Algebra.smul_def]
+  rfl
+
+/-- If `A` and `L` are linearly disjoint, one of them is algebraic, then `[L(A) : A] = [L : F]`.
+Note that in Lean `L(A)` is not naturally an `A`-algebra,
+so this result is stated in a cumbersome way. -/
+theorem lift_adjoin_rank_eq_lift_rank_right_of_isAlgebraic (H : A.LinearDisjoint L)
+    (halg : Algebra.IsAlgebraic F A ∨ Algebra.IsAlgebraic F L) :
+    Cardinal.lift.{w} (Module.rank A (extendScalars
+      (show A ≤ (adjoin L (A : Set E)).restrictScalars F from subset_adjoin L (A : Set E)))) =
```

Comment:
Is `adjoin L (A : Set E)` really the better spelling here instead of `L ⊔ A`?

Reply:
`L ⊔ A` doesn't typecheck because L is not an intermediate field ... it's only assumed that E/L/F is a tower.

Reply:
I should say this particular result is for advanced users only. Regular user should use `adjoin_rank_eq_rank_left_of_isAlgebraic` (`[L(A) : L] = [A : F]`) and `lift_rank_right_mul_lift_adjoin_rank_eq_of_isAlgebraic` (`[L : F] * [E : L(A)] = [E : A]`) and which are the downstream applications in this PR.